zerver/management/commands/reupload_emoji.py

```diff
@@ -0,0 +1,77 @@
+import io
+from argparse import ArgumentParser
+from typing import Any
+from urllib.parse import urljoin
+
+from django.core.management import CommandError
+
+from zerver.lib.emoji import get_emoji_url
+from zerver.lib.management import ZulipBaseCommand
+from zerver.lib.outgoing_http import OutgoingSession
+from zerver.lib.upload import upload_emoji_image
+from zerver.models import RealmEmoji, UserProfile
+
+
+def get_emoji_file_content(session: OutgoingSession, emoji_url: str, emoji_id: int) -> bytes:
```

Comment:
I am a bit concerned with the use of requests to fetch the emoji data, but it might be because I'm not familiar enough with the architecture. 

How do we know that the webserver will be running and reachable by this script? Custom installs might do weird stuff with proxies/nginx config. Also imagine you've deployed using Docker or a PAAS like heroku. Usually the shell you get (to run management commands) would be an instance of the container, not inside the container itself. In that case there might not be any webserver running and available locally at all.

Reply:
I think this is a valid concern but I don't think there's a convenient alternative generic way to access the contents of uploaded files.

Probably this belongs in the documentation for using this tool -- you should run it while your Zulip server is operational.

Reply:
Yeah, I've added an explicit note that the server should be running to the upgrade notes and the help string of the command, good point. 

>  I don't think there's a convenient alternative generic way to access the contents of uploaded files.

@timabbott #20737 uses backend-specific code to fetch the files natively instead of this http request approach, but it gets more messy, so not sure if it's worth it